src/goto-symex/symex_builtin_functions.cpp

```diff
@@ -135,14 +135,15 @@ void goto_symext::symex_allocate(
     {
       exprt &size=to_array_type(object_type).size();
 
-      symbolt size_symbol;
+      auxiliary_symbolt size_symbol;
 
       size_symbol.base_name=
         "dynamic_object_size"+std::to_string(dynamic_counter);
       size_symbol.name="symex_dynamic::"+id2string(size_symbol.base_name);
-      size_symbol.is_lvalue=true;
       size_symbol.type=tmp_size.type();
       size_symbol.mode=ID_C;
+      size_symbol.type.set(ID_C_constant, true);
```

Comment:
Same as above, while at it!

Reply:
I'm not sure what I should make the same as above? I can't use a member of `c_typecheck_baset` here?!

Reply:
Sorry, I should be picking up the fix that @peterschrammel has provided and use `mode`, understood.